src/Modules/ModulesGraded.jl

```diff
@@ -0,0 +1,448 @@
+export FreeMod_dec, FreeModElem_dec, decoration
+
+abstract type ModuleFP_dec{T} <: ModuleFP{T} end
+abstract type AbstractFreeMod_dec{T} <: AbstractFreeMod{T} end
+abstract type AbstractSubQuo_dec{T} <: AbstractSubQuo{T} end
+
+abstract type AbstractFreeModElem_dec{T} <: AbstractFreeModElem{T} end
+abstract type AbstractSubQuoElem_dec{T} <: AbstractSubQuoElem{T} end
+
+const CRing_dec = Union{MPolyRing_dec, MPolyQuo{<:Oscar.MPolyElem_dec}}
+const CRingElem_dec = Union{MPolyElem_dec, MPolyQuoElem{<:Oscar.MPolyElem_dec}}
+#TODO: other name for CRing_dec -> which?
+
+
+@doc Markdown.doc"""
+    FreeMod_dec{T <: CRingElem_dec} <: AbstractFreeMod_dec{T}
+
+The type of decorated (graded or filtered) free modules.
+Decorated free modules are determined by their base ring, the rank,
+the grading or filtration and the names of the (standard) generators.
+Moreover, canonical incoming and outgoing morphisms are stored if the corresponding
+option is set in suitable functions.
+`FreeMod_dec{T}` is a subtype of `ModuleFP{T}`.
+"""
+mutable struct FreeMod_dec{T <: CRingElem_dec} <: AbstractFreeMod_dec{T}
+  F::FreeMod{T}
+  d::Vector{GrpAbFinGenElem}
+
+  AbstractAlgebra.@declare_other
```

Comment:
```suggestion
@attributes mutable struct FreeMod_dec{T <: CRingElem_dec} <: AbstractFreeMod_dec{T}
  F::FreeMod{T}
  d::Vector{GrpAbFinGenElem}
```
This PR needs to be updated to use `get_attribute` etc. instead of `get_special` etc., see https://github.com/Nemocas/AbstractAlgebra.jl/pull/1065